src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -138,7 +138,10 @@ public JsonClassInfo(Type type, JsonSerializerOptions options)
                     {
                         CreateObject = options.MemberAccessorStrategy.CreateConstructor(type);
 
-                        PropertyInfo[] properties = type.GetProperties(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic);
```

Comment:
I believe the only fix here is to remove the `BindingFlags.NonPublic` flag. It's not conveying the intent of the serializer and likely just an oversight (maybe copy/paste error). In reality, non-public properties are **not** supported.

Reply:
I believe `NonPublic` is necessary to support public getters and non-public setters.

Reply:
Responded in the PR that's removing it: https://github.com/dotnet/runtime/pull/2278#issuecomment-580388963